boringtun/src/device/mod.rs

```diff
@@ -416,25 +416,23 @@ impl Device {
         }
 
         if let Some(peer) = self.peers.get(&pub_key) {
-            let mut p = peer.lock();
-
             if let Some(endpoint) = endpoint {
-                p.set_endpoint(endpoint);
+                peer.set_endpoint(endpoint);
             }
 
             if replace_ips {
                 self.peers_by_ip.remove(&|p| Arc::ptr_eq(&peer, p));
-                p.set_allowed_ips(&allowed_ips);
+                peer.set_allowed_ips(&allowed_ips);
             } else {
-                p.add_allowed_ips(&allowed_ips);
+                peer.add_allowed_ips(&allowed_ips);
```

Comment:
Minor: in diffs such as these with a lot of uniform changes it would be nice to reuse the variables to reduce the diff, thus here it could still be `p`. Very easy to miss on the meaning of the PR :)